src/types.ts

```diff
@@ -31,12 +31,12 @@ type ListenerObject<TEvent extends Event> = {
 };
 
 // event listeners can be an object or a function
-export type Listener<TTarget extends EventTarget, TEvent extends Event> =
-  | ListenerObject<TEvent>
-  | { (this: TTarget, ev: TEvent): void };
+export type Listener<TTarget extends EventTarget, TType extends string = string> =
```

Comment:
@Andarist I am keen for your thoughts here. I

 moved `Listener` back to taking a `string` rather than an `Event` in order to maintain the previous type signature. Autocomplete seems to be continuing to work great.

Reply:
After this change the only _breaking_ change would be the type signature of `bindAll`

Reply:
Or this is bad because we have separate calculating of what `TEvent` is, rather that a single inference at the root? 🤔